src/Stratis.Bitcoin.Features.Wallet/Controllers/MultisigController.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Features.Wallet.Interfaces;
+using Stratis.Bitcoin.Features.Wallet.Models;
+using Stratis.Bitcoin.Utilities;
+using Stratis.Bitcoin.Utilities.JsonErrors;
+using Stratis.Bitcoin.Utilities.ModelStateErrors;
+
+namespace Stratis.Bitcoin.Features.Wallet.Controllers
+{
+    /// <summary>
+    /// Controller providing operations on a wallet.
+    /// </summary>
+    [Route("api/[controller]")]
+    public class MultisigController : Controller
+    {
+        private readonly IMultisigTransactionHandler multisigTransactionHandler;
+
+        /// <summary>Specification of the network the node runs on - regtest/testnet/mainnet.</summary>
+        private readonly Network network;
+
+        /// <summary>Instance logger.</summary>
+        private readonly ILogger logger;
+
+        public MultisigController(
+            ILoggerFactory loggerFactory,
+            IMultisigTransactionHandler multisigTransactionHandler,
+            Network network)
+        {
+            this.multisigTransactionHandler = multisigTransactionHandler;
+            this.network = network;
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
+        }
+
+        /// <summary>
+        /// Builds a transaction and returns the hex to use when executing the transaction.
+        /// </summary>
+        /// <param name="request">An object containing the parameters used to build a transaction.</param>
+        /// <returns>A JSON object including the transaction ID, the hex used to execute
+        /// the transaction, and the transaction fee.</returns>
+        [Route("build-transaction")]
+        [HttpPost]
+        public IActionResult BuildTransaction([FromBody] BuildMultisigTransactionRequest request)
+        {
+            Guard.NotNull(request, nameof(request));
+
+            if (!this.ModelState.IsValid)
+                return ModelStateErrors.BuildErrorResponse(this.ModelState);
+
+            try
+            {
+                var recipients = new List<Recipient>();
+                foreach (RecipientModel recipientModel in request.Recipients)
+                {
+                    recipients.Add(new Recipient
+                    {
+                        ScriptPubKey = BitcoinAddress.Create(recipientModel.DestinationAddress, this.network).ScriptPubKey,
+                        Amount = recipientModel.Amount
+                    });
+                }
+
+                var context = new TransactionBuildContext(this.network)
+                {
+                    AccountReference = new WalletAccountReference(request.WalletName, request.AccountName),
```

Comment:
In the context of building the transaction to move the funds, what will the wallet and account name's be? Because we aren't moving funds from a "normal" wallet but from a multi_sig one. So when we build the transaction with transaction builder it won't know where to get the UTXOs from. 

Or does it actually matter?